src/modalities/trainer.py

```diff
@@ -101,9 +103,7 @@ def train(
             # Check, if model should be evaluated
             if (local_train_batch_id + 1) % callback_interval_in_batches == 0:
                 if local_train_batch_id > 0:
-                    foward_backward_time = torch.tensor(forward_backward_time_recorder.delta_t).to(
-                        torch.device(self.local_rank)
-                    )
+                    foward_backward_time = torch.tensor(forward_backward_time_recorder.delta_t).to(device)
                     forward_backward_time_recorder.reset()
 
                     thoughput_aggregator.add_value(
```

Comment:
The previously used mocking of the "reduce" operation led to problems below. More concretely, line 112 and line 115 both returned `None` (the `return_value`), which caused an error in line 118.